### ISSUE TITLE
Transform question and issues

### ISSUE THREAD

**Issue by DaviehS** [taskrabbit/elasticsearch-dump#620]
First of all, I can't get transform to work according to the [wiki ](https://www.npmjs.com/package/elasticdump#module-transform), I created a folder named `transforms` inside of it a file named `my-transform.js` and the code is :
```
module.exports = function (doc, options) {
    // do something to doc
   doc._source = "aa"
	return doc
};
```
command ` elasticdump --type=data --input=XXXX --output=./data_7_23.json transform='@./transforms/my-transform?index=company-firewall-2018.06.10'` (I tried adding .js to the filename still nothing)
Nothing happens - I tried different js code, it literally ignores it.

But when I specify a direct js code in the command it works, for example 
`elasticdump --type=data --input=XXX --output=./data_7_231.json --transform='doc._source = "a"'`

Second and more important question, is how can I modify the whole `doc` object? I was only able to modify fields in it, for example `doc._source`, but not doc itself.

I'm trying to form the output to be compatible with ES Bulk api, where as every doc have it's meta_data header first, and then the doc itself. I'm aware of the option `--sourceOnly` which outputs only the `_source`

Is there a way I can Format my output in the following way :

```
{ "index" : { "_index" : "test", "_id" : "1" } }
{ "field1" : "value1" }
```

basically break down the normal output into 2 "docs"\lines

Thank you very much!

**Comment by ferronrsmith** [comment]
1. The input is incorrect. It needs to be : 

elasticdump --type=data --input=XXXX --output=./data_7_23.json transform=@./transforms/my-transform?index=company-firewall-2018.06.10

> **Do not wrap the quotes**

2. The entire object is passed into the transform. You modify it, as you would any other object. An example is provided here : https://github.com/taskrabbit/elasticsearch-dump/blob/master/transforms/anonymize.js

**Comment by DaviehS** [comment]
@ferronrsmith 
1. I tried that previously also, removing the quotes, still nothing happens!
`elasticdump --type=data --input=XXX --output=./data_7_231.json transform=@./transforms/my-transform?index=company-firewall-2018.06.10` 
when I use inplace command then is says 
`| Will modify documents using these scripts: return doc._source.profile = "a"`
for example, when i try using a `.js` file, it doesn't say that.

2. I understand I can modify, but can i just make doc as a string ? this way I can have 2 files inside one file, otherwise if it's a json, I can't apply the bulk api format of 2 jsons together
Or is there another way?
Your answer doesn't tell me anything I didn't know, I'm asking because I cannot accomplish what I explained in my original post, if you can please help me ....

**Comment by ferronrsmith** [comment]
1. I tried it a while ago and it work for me. Maybe the location you're loading is incorrect. I tried it and it worked for me. Try using a absolute path.
2. no

**Comment by DaviehS** [comment]
ok, thank you!

I'll parse the `sourceOnly` and add to it manually a header for each row.
